runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -2202,3 +2202,72 @@ TR_J9SharedCacheServerVM::getClassFromNewArrayType(int32_t arrayType)
       return TR_J9ServerVM::getClassFromNewArrayType(arrayType);
    return NULL;
    }
+
+void *
+TR_J9SharedCacheServerVM::getJ2IThunk(char *signatureChars, uint32_t signatureLength, TR::Compilation *comp)
+   {
+   return (void *)findPersistentThunk(signatureChars, signatureLength);
+   }
+
+void *
+TR_J9SharedCacheServerVM::setJ2IThunk(char *signatureChars, uint32_t signatureLength, void *thunkptr, TR::Compilation *comp)
+   {
+   return persistThunk(signatureChars, signatureLength, (U_8*)thunkptr - 8 /* start of thunk */, *((U_32 *)((U_8*)thunkptr -8)) + 8 /* size of thunk */);
```

Comment:
This code dereferences a pointer (thunkptr-8). Does the information reside at the server?

Reply:
Thunks are generated during codegen on the server, so dereferencing it should be fine.